logstreams/src/main/java/io/zeebe/logstreams/impl/log/LogStreamReaderImpl.java

```diff
@@ -0,0 +1,278 @@
+  public boolean seekToNextEvent(final long position) {
+
+    if (position <= -1) {
+      seekToFirstEvent();
+      return true;
+    }
+
+    final boolean found = seek(position);
+    if (found && hasNext()) {
+      next();
+      return true;
+    }
+
+    return false;
+  }
+
+  @Override
+  public boolean seek(final long position) {
+    if (state == IteratorState.CLOSED) {
+      throw new IllegalStateException(ERROR_CLOSED);
+    }
+
+    final long seekAddress = storageReader.lookUpApproximateAddress(position);
+    invalidateBufferAndOffsets();
+    return seekFrom(seekAddress, position);
+  }
+
+  @Override
+  public void seekToFirstEvent() {
+    seek(FIRST_POSITION);
+  }
+
+  @Override
+  public long seekToEnd() {
+    // invalidate events first as the buffer content may change
+    invalidateBufferAndOffsets();
+
+    final long blockAddress = storageReader.getFirstBlockAddress();
+    if (blockAddress < 0) {
+      // no block found => empty log
+      state = IteratorState.EMPTY_LOG_STREAM;
+      return -1;
+    } else {
+      if (readLastBlockIntoBuffer()) {
+        do {
+          nextEvent.wrap(nextEventBuffer, bufferOffset);
+          bufferOffset += nextEvent.getLength();
+        } while (bufferOffset < nextEventBuffer.capacity());
+
+        return nextEvent.getPosition();
+      }
+
+      // if the log is not empty this should not happen however
+      Loggers.LOGSTREAMS_LOGGER.warn("Unexpected non-empty log failed to read the last block");
+      return UNINITIALIZED;
+    }
+  }
+
+  @Override
+  public long getPosition() {
+    // if an event was already returned use it's position otherwise use position of next event if
+    // available, kind of strange but seemed to be the old API
+    if (isReturnedEventInitialized()) {
+      return returnedEvent.getPosition();
+    }
+
+    if (state == IteratorState.EVENT_AVAILABLE) {
+      return nextEvent.getPosition();
+    }
+
+    return UNINITIALIZED;
+  }
+
+  @Override
+  public long lastReadAddress() {
+    return lastReadAddress;
+  }
+
+  private boolean seekFrom(final long blockAddress, final long position) {
+    if (blockAddress < 0) {
+      // no block found => empty log
+      state = IteratorState.EMPTY_LOG_STREAM;
+      return false;
+    } else {
+      readBlockIntoBuffer(blockAddress);
+      readNextEvent();
+      return searchPositionInBuffer(position);
+    }
+  }
+
+  @Override
+  public void close() {
+    nextEventBuffer.wrap(0, 0);
+    bufferOffset = 0;
+    state = IteratorState.CLOSED;
+
+    if (storageReader != null) {
+      storageReader.close();
+      storageReader = null;
+    }
+  }
+
+  @Override
+  public boolean hasNext() {
+    switch (state) {
+      case EVENT_AVAILABLE:
+        return true;
+      case EMPTY_LOG_STREAM:
+        seekToFirstEvent();
+        break;
+      case NOT_ENOUGH_DATA:
+        readNextAddress();
+        break;
+      case EVENT_NOT_COMMITTED:
+        state = IteratorState.EVENT_AVAILABLE;
```

Comment:
How we can say this?

Reply:
That was already like this :sweat_smile: I think the logic was that since we can __only__ read committed events, we assume they are always committed? Actually I think we can remove this since it's useless now

Reply:
Yes lets remove it